generators/server/templates/_pom.xml

```diff
@@ -974,9 +977,6 @@
                     Enable the line below to have remote debugging of your application on port 5005
                     <jvmArguments>-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005</jvmArguments>
                     -->
-                    <arguments>
```

Comment:
@deepu105 I am finding the same thing. I didn't notice earlier, because we use H2 for production as well.

You are right that adding 
```xml
<arguments>
    <argument>spring.profiles.active=${spring.profiles.active}</argument>
</arguments>
```
to the spring-boot plugin configuration "works". But it seems to me that this is just suppressing the symptom of a deeper problem. 

It looks to me like this is a chicken/egg situation around the order in which:

-- Spring looks to VM arguments for explicit profiles set to active
-- Database config looks for properties from active profiles
-- Spring looks at application.yml for active profiles

I saw similar weirdness when working on the original default profile stuff, which was why my original effort ended up using a static initializer block (in BuildProperties.java) to make sure that the default profile was set before anything else happened.

I am curious if this is because the DB configuration uses annotations like `@ConfigurationProperties` and `@Profile`. Would it maybe help to just check for profile dynamically instead?